frontend/src/components/organisms/SignupForm.tsx

```diff
@@ -6,127 +6,198 @@ import { useAuth } from "@/utils/AuthContext";
-            label="First Name *"
-            required={true}
-          />
-        </div>
-        <div>
-          <TextField
-            requiredMessage={errors.lastName ? "Required" : undefined}
-            name="lastName"
-            register={register}
-            label="Last Name *"
-            required={true}
-          />
-        </div>
-      </div>
-      <div>
-        <TextField
-          requiredMessage={errors.password ? "Required" : undefined}
-          type="password"
-          name="password"
-          register={register}
-          label="Password *"
-          required={true}
-        />
-      </div>
-      <div>
-        <TextField
-          type="password"
-          requiredMessage={
-            errors.confirmPassword
-              ? "Required"
-              : watch("password") != watch("confirmPassword")
-              ? "Passwords do not match"
-              : undefined
-          }
-          name="confirmPassword"
-          register={register}
-          label="Confirm Password *"
-          required={true}
-        />
-      </div>
-      <div>
-        <Button color="dark-gray" type="submit">
-          Continue
-        </Button>
-      </div>
-      <div>
-        <Button type="submit" color="gray">
-          Continue with Google
-        </Button>
-      </div>
-      <div className="justify-center flex flex-row">
-        <div className="">Have an account?&nbsp;</div>
-        <Link href="/login" className="text-black">
-          {" "}
-          Log in
-        </Link>
-      </div>
-    </form>
-  );
+	const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+	const handleErrors = (errors: any) => {
+		const userAlreadyExistsPrisma = /Unique constraint failed/;
+
+		if (userAlreadyExistsPrisma.test(errors)) {
+			return "A user with that email already exists.";
+		} else if (errors == "Passwords do not match") {
+			return "Passwords do not match";
+		}
+
+		// Firebase errors
+		switch (errors) {
+			case "auth/email-already-exists":
+				return "A user with that email already exists.";
+			case "auth/invalid-email":
+				return "Invalid email address format.";
+			case "auth/invalid-password":
+				return "Password should be at least 6 characters.";
+			default:
+				return "Something went wrong trying to create your account. Please try again.";
+		}
+	};
+
+	const SignUpErrorComponent = (): JSX.Element => {
+		return (
+			<div>
+				{errorMessage ? (
+					<CustomAlert
+						severity="error"
+						title="Error"
+						message={handleErrors(errorMessage)}
+					/>
+				) : null}
+			</div>
+		);
+	};
+
+	const handleSubmitUser: SubmitHandler<FormValues> = async (data, event) => {
+		setIsLoading(true);
+
+		event?.preventDefault();
+		const { firstName, lastName, email, password } = data;
+		const emailLowerCase = email.toLowerCase();
+		const post = {
+			email: emailLowerCase,
+			profile: {
+				firstName,
+				lastName,
+			},
+			password,
```

Comment:
Actually wait yeah sending passwords through `POST` is not secure at all

Reply:
I did some more digging and this should be fine as long as we use `https` instead of `http`, which ensures that a secure encrypted SSL tunnel is established before packets are sent.

Reply:
Yep. As long as it is over https, it is encrypted over the wire